main.go

```
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+)
+
+type response struct {
+	Host       string      `json:"host"`
+	UserAgent  string      `json:"user_agent"`
+	RequestUri string      `json:"request_uri"`
+	Header     http.Header `json:"headers"`
+}
+
+func handleRoot(w http.ResponseWriter, r *http.Request) {
+	responseJSON, err := json.Marshal(response{
+		r.Host,
+		r.Header.Get("User-Agent"),
+		r.RequestURI,
+		r.Header,
+	})
+	if err != nil {
+		fmt.Println(err)
```

Please return an error and error code to client instead.